src/sentry/stacktraces/processing.py

```diff
@@ -178,41 +178,48 @@ def preprocess_step(self, processing_task):
 
 
 def find_stacktraces_in_data(
-    data: NodeData, include_raw: bool = False, with_exceptions: bool = False
+    data: NodeData,
+    include_raw: bool = False,
+    with_exceptions: bool = False,
+    platform: str = "unknown",
 ) -> list[StacktraceInfo]:
     """Finds all stacktraces in a given data blob and returns it
     together with some meta information.
 
-    If `include_raw` is True, then also raw stacktraces are included.  If
-    `with_exceptions` is set to `True` then stacktraces of the exception
+    If `include_raw` is True, then also raw stacktraces are included.
+
+    If `with_exceptions` is set to `True` then stacktraces of the exception
     are always included and the `is_exception` flag is set on that stack
     info object.
     """
     rv = []
+    _platform = data.get("platform", platform)
 
     def _append_stacktrace(stacktrace: Any, container: Any, is_exception: bool = False) -> None:
         if not is_exception and (not stacktrace or not get_path(stacktrace, "frames", filter=True)):
             return
 
-        platforms = {
-            frame.get("platform") or data.get("platform")
-            for frame in get_path(stacktrace, "frames", filter=True, default=())
-        }
-        rv.append(
-            StacktraceInfo(
-                stacktrace=stacktrace,
-                container=container,
-                platforms=platforms,
-                is_exception=is_exception,
+        frames = extract_structure_from_json_structure(stacktrace, "frames")
```

Comment:
The `extract_structure_from_json_structure` is a wrapper to `get_path` which is easier to understand for someone reading the code.